File: python/414 - Third Maximum Number.py
class Solution(object):
    def thirdMax(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums = list(set(nums))
        nums.sort()
        n = len(nums)
        
        if n >= 3:
            return nums[n - 1 - 2]
        else:
            return nums[n - 1]

File: cpp/414 - Third Maximum Number.cpp
class Solution {
public:
    int thirdMax(vector<int>& nums) {
        set<int> s(nums.begin(), nums.end());
        vector<int> sorted(s.begin(), s.end());
        sort(sorted.begin(), sorted.end());
       
        int n = sorted.size();
        if (n >= 3)
            return sorted[n - 1 - 2];
        else
            return sorted[n - 1];
    }
};